pkg/app/piped/executor/kubernetes/canary_test.go

```diff
@@ -175,11 +175,13 @@ func TestEnsureCanaryRollout(t *testing.T) {
 					}, nil)
 					return p
 				}(),
-				applier: func() provider.Applier {
-					p := kubernetestest.NewMockApplier(ctrl)
-					p.EXPECT().ApplyManifest(gomock.Any(), gomock.Any()).Return(fmt.Errorf("error"))
-					return p
-				}(),
+				applierGetter: &applierGroup{
```

Comment:
How about creating a mock for applierGetter?

Reply:
We are checking the call of an internal object (default applier), so I think mock for applierGetter could not be helpful in this case.
Please correct me if I'm wrong.

Reply:
IMO, If you want to test whether the interface of applierGetter calls some methods or not, we should write tests for applierGroup to check it.
But I could also understand what you meant, it's up to you. 🙌

Reply:
I got your point. You meant this test should only care about the applierGetter interface. If we want to test its internal calls that should be done in other tests, right?

Reply:
@knanao OK. Let me do it together with this TODO https://github.com/pipe-cd/pipecd/pull/3817#issuecomment-1192405511 in another PR.

Reply:
Thank you 🙌